src/Cassandra/Session.cs

```diff
@@ -107,6 +107,17 @@ public void ChangeKeyspace(string keyspace)
         }
 
         /// <inheritdoc />
+        public Task ChangeKeyspaceAsync(string keyspace)
+        {
+            if (Keyspace != keyspace)
+            {
+                var task = ExecuteAsync(new SimpleStatement(CqlQueryTools.GetUseKeyspaceCql(keyspace)));
+                return task.Continue(rs => Keyspace = keyspace);
```

Comment:
You can safely remove the setting of the keyspace as this is done by `RequestExecution` class, after a [`set_keyspace` response](https://git-wip-us.apache.org/repos/asf?p=cassandra.git;a=blob_plain;f=doc/native_protocol_v4.spec;hb=refs/heads/cassandra-2.2).
This can be `return ExecuteAsync(...`

It is also not necessary on the [sync counterpart](https://github.com/datastax/csharp-driver/pull/161/files#diff-5137a268854a1ede5dc7e5c2ef5c5d3dL105), but was left there from a refactoring